src/signature/mechanism/cosign/mod.rs

```diff
@@ -173,8 +177,18 @@ impl CosignParameters {
             }) => Err(anyhow!("{:?}", unsatisfied_constraints)),
         }
     }
+
+    #[cfg(target_arch = "s390x")]
+    async fn verify_signature_and_get_payload(
+        &self,
+        _image: &mut Image,
+        _auth: &RegistryAuth,
+    ) -> Result<Vec<SigPayload>> {
+        Err(anyhow!("Cosign is not supported on IBM s390x"))
```

Comment:
Could you add a comment with referencing to the issue to let the reader understand why and that there is work in progress to support it (see ring discussions).

Reply:
I added a comment about big-endian architectures being not currently supported by ring, with a link to the ring issue for adding support for big-endian architectures including s390x.

Reply:
Thanks you!